pkg/destroy/azure/azure.go

```diff
@@ -629,5 +633,9 @@ func getServicePrincipalsByTag(ctx context.Context, graphClient *msgraphsdk.Grap
 	if err != nil {
 		return nil, err
 	}
+	// msgraphsdk can return a `nil` response even if no errors occurred
+	if resp == nil {
```

Comment:
nit: Since the response can be nil, couldn't we simplify the return to 

```
if err != nil || resp == nil {
    return nil, err
}
```

Reply:
Indeed. Will do.

Reply:
We only want to return `err` if it is not nil. As the code stands now, if `resp != nil` but `err == nil` we would return the nil err and calling code would panic.